python/lsst/ts/ATWhiteLightSource/wlsCSC.py

```diff
@@ -643,15 +659,22 @@ async def telemetryLoop(self):
 
             await asyncio.sleep(self.telemetry_publish_interval)
 
-    async def close(self):
-        self.interlockLoopBool = False
-        self.telemetryLoopBool = False
-        self.kiloarcListenerLoopBool = False
-        self.kiloarcModel.disconnect()
-        await self.chillerModel.disconnect()
-        await self.kilo_interloc
-        await self.telemetryLoopTask
-        await self.kiloarcListenerTask
+    async def close_tasks(self):
+        self.log.debug("Running csc close_tasks method")
+        await self._common_close()
         self.kiloarcModel.cooldown_task.cancel()
         self.kiloarcModel.warmup_task.cancel()
-        await super().close()
+        await super().close_tasks()
+
+    async def _common_close(self):
+        """
+        closes the high level tasks and disconnects from models, but doesn't
+        kill the safety timers
+        """
+        self.telemetryLoopTask.cancel()
+        self.interlock_task.cancel()
+        self.kiloarcListenerTask.cancel()
+        if self.kiloarcModel is not None:
+            self.kiloarcModel.disconnect()
```

Comment:
I strongly suggest making the kiloarcModel connect and disconnect methods asynchronous. This has several advantages:
* It keeps your code self-consistent and thus more predictable. Both the chiller and kiloarc models have the same API for connection and disconnection. which makes your code more readable (a big win).
* If you ever do need to await something you don't need to change the API. Note that connection often can block, so being able to await is a very natural thing for a connect method.